mogwai-schedule-client/main.c

```diff
@@ -829,8 +829,16 @@ notify_cb (GObject    *obj,
            GParamSpec *pspec,
            gpointer    user_data)
 {
-  g_autofree gchar *message = g_strdup_printf (_("%s was notified"),
-                                               g_param_spec_get_name (pspec));
+  MwscScheduler *scheduler = MWSC_SCHEDULER (obj);
+  const gchar *property_name = g_param_spec_get_name (pspec);
+  g_autofree gchar *message = NULL;
+
+  if (g_str_equal (property_name, "allow-downloads"))
+    message = g_strdup_printf (_("%s changed to %s"), property_name,
+                               mwsc_scheduler_get_allow_downloads (scheduler) ? "yes" : "no");
+  else
+    message = g_strdup_printf (_("%s changed value"), property_name);
+
```

Comment:
I am not a big fan of allocating memory just for logging reasons (unless there's no good alternative). In this case I'd prefer to repeat the `g_print` call with the different messages.

Reply:
Allocation is necessary to avoid having 3 different translatable strings with basically the same content.

Also, these messages get emitted really really rarely (like, only when you change network connection). It’s the antithesis of performance critical.

Reply:
@pwithnall I really don't understand how you avoid "3 different translated strings" by allocating memory here.
Wouldn't you get the same (minus mem allocation) if you called:
```
if (g_str_equal (property_name, "allow-downloads"))
  g_print (_("%s changed to %s"), ...);
else
  g_print (_("%s changed value"), ...);
```
Or what am I missing?

Reply:
Ah, I see where you’re coming from. The `g_print()` is split from the `g_strdup_printf()` so that the trailing `\n` isn’t in the translatable message, since translators sometimes mess up the escaping/delete the newline/try and translate it, and it should really be independent from translation anyway.